File: queue/Queue.java
import java.util.ArrayList;
import java.util.List;

public class Queue<T> {
    private List<T> queue = new ArrayList<T>();

    public void push(T value) {
        queue.add(value);
    }

    public T pop() {
        T value = queue.get(0);
        queue.remove(0);
        return value;
    }

    public int size() {
        return queue.size();
    }

    public T front() {
        return queue.get(0);
    }

    public T back() {
        return queue.get(size() - 1);
    }

    public boolean isEmpty() {
        return queue.isEmpty();
    }

    public static void main(String[] args) {
        var queue = new Queue<Integer>();

        for (int i = 0; i < 10; i++) {
            queue.push(i);
        }

        System.out.println(queue.back());

        while (!queue.isEmpty()) {
            if (queue.front() % 2 == 0) {
                System.out.println(queue.front());
            }
            queue.pop();
        }
    }
}

File: queue/queue.py
class Queue:
    def __init__(self):
        self.__queue = []
    
    def push(self, value):
        self.__queue.append(value)
    
    def pop(self):
        return self.__queue.pop(0)
    
    def size(self):
        return len(self.__queue)

    def is_empty(self):
        return self.size() == 0;

    def front(self):
        return self.__queue[0]

    def back(self):
        return self.__queue[-1]


if __name__ == '__main__':
    elements = [i for i in range(10)]
    queue = Queue()
    
    for element in elements:
        queue.push(element)
    
    print(queue.back())
    
    while not queue.is_empty():
        if queue.front() % 2 == 0:
            print(queue.front())
        queue.pop()
